evm/its.js

```diff
@@ -0,0 +1,723 @@
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowLimit = await interchainTokenService.flowLimit(tokenIdBytes32);
+            printInfo(`Flow limit for TokenManager with tokenId: ${tokenId}`, flowLimit);
+
+            break;
+        }
+
+        case 'flowOutAmount': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowOutAmount = await interchainTokenService.flowOutAmount(tokenIdBytes32);
+            printInfo(`Flow out amount for TokenManager with tokenId: ${tokenId}`, flowOutAmount);
+
+            break;
+        }
+
+        case 'flowInAmount': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const flowInAmount = await interchainTokenService.flowInAmount(tokenIdBytes32);
+            printInfo(`Flow out amount for TokenManager with tokenId: ${tokenId}`, flowInAmount);
+
+            break;
+        }
+
+        case 'deployTokenManager': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { salt, destinationChain, type, params, gasValue } = options;
+
+            if (!isKeccak256Hash(salt)) {
+                throw new Error(`Invalid salt: ${salt}`);
+            }
+
+            if (!isString(destinationChain)) {
+                throw new Error(`Invalid destinationChain: ${destinationChain}`);
+            }
+
+            if (!isValidCalldata(params)) {
+                throw new Error(`Invalid params: ${params}`);
+            }
+
+            if (!isValidNumber(gasValue)) {
+                throw new Error(`Invalid gas value: ${gasValue}`);
+            }
+
+            const tx = await interchainTokenService.deployTokenManager(
+                salt,
+                destinationChain,
+                tokenManagerImplementations[type],
+                params,
+                gasValue,
+            );
+            printInfo('deploy TokenManager tx', tx.hash);
+
+            const receipt = await tx.wait(chain.confirmations);
+
+            const eventEmitted =
+                wasEventEmitted(receipt, interchainTokenService, 'TokenManagerDeployed') ||
+                wasEventEmitted(receipt, interchainTokenService, 'TokenManagerDeploymentStarted');
+
+            if (!eventEmitted) {
+                printWarn('Event not emitted in receipt.');
+            }
```

Comment:
if it was a tx, then wait + event check can be done generically at the end for all cases, event to be checked can be defined here

Reply:
moved this check to a separate function